src/openrct2/CmdlineSprite.cpp

```diff
@@ -183,7 +184,62 @@ bool SpriteFile::Save(const utf8* path)
     }
 }
 
-static bool SpriteImageExport(const rct_g1_element& spriteElement, const char* outPath)
+static std::optional<GamePalette> PaletteImageImport(const char* path)
+{
+    try
+    {
+        auto format = IMAGE_FORMAT::PNG_32;
+        auto image = Imaging::ReadFromFile(path, format);
+
+        auto width = image.Width;
+        auto height = image.Height;
+        if (width * height != 256)
+        {
+            throw "Custom palette image is not 256 pixels";
+        }
+        GamePalette newPalette;
+        auto pixels = image.Pixels.data();
+        auto src = pixels;
+        auto palIndex = 0;
+        for (uint32_t y = 0; y < height; y++)
```

Comment:
the image is height*width we don't need to mess about with any of this you can just do `for (int i = 0; i < 256; ++i)` stride doesn't matter.